tutorial/Tutorial.md

```diff
@@ -0,0 +1,128 @@
+
+In this tutorial we are showing a summary of a workflow with Move on a Substrate template node. We will publish a module and execute a script which uses that module's functionality. Therefore, the package manager `smove` will be used to compile those resources, to estimate the needed amount of gas, and to create the script-transaction. Finally we publish the module and execute a script via using [polkadot.js](https://polkadot.js.org/apps/).
+
+
+## Setup and Code Example 
+
+Initial requirements:
+- [Install smove](https://github.com/eigerco/smove)
+- Setup a substrate node with pallet-move integrated or use [our template node](https://github.com/eigerco/substrate-node-template-move-vm-test)
+- Switch into the code example directory in `pallet-move/tests/assets/move-projects/car-wash-example`
+
+In our example, a simple car wash is modelized, where washing coins are used to control the usage of the car wash. Users can register, buy washing coins and use them to start the car wash. For the sake of simplicity, we show here only the function headers of the Move module, the full module can be seen [here](https://github.com/eigerco/pallet-move/blob/main/tests/assets/move-projects/car-wash-example/sources/CarWash.move).
+
+```move
+module DeveloperBob::CarWash {
+    /// Struct stores number of coins for each user.
+    struct Balance has key, store {
+        coins: u8
+    }
+
+    /// Method executes the ICO without money. The module owner (also the car wash owner) gets deposited the minted washing coins.
+    public fun initial_coin_minting(module_owner: &signer) {}
+
+    /// Registers a new user. The account address will be added to the storage Balance with zero initial washing coins.
+    public fun register_new_user(account: &signer) {}
+
+    /// Buys a washing coin for the car wash. Therfore, `COIN_PRICE` will be withdrawn from the user's account.
+    // Note: It would be nice to have the ability to buy multiple coins at once, but it's not implemented for this example.
+    public fun buy_coin(user: &signer) acquires Balance {}
+
+    /// Initiates the washing process by paying one washing coin.
+    public fun wash_car(user: &signer) acquires Balance {}
+}
+```
+
+Each publically usable method will be depicted into an executable script, like this style:
+```move
+script {
+    use DeveloperBob::CarWash;
+    
+    fun initial_coin_minting(account: signer) {
+        CarWash::initial_coin_minting(&account);
+    }
+}
+```
+
+Now let's compile this project to be ready for estimation of needed gas and publication of the module.
+```sh
+smove build
+```
+
+## Estimation of Needed Gas
+
+Gas can be seen as weights in Substrate. It will adjust the fees for the execution or publication of Move resources. Hereby we prevent that scripts will be sued to exploit somehow the blockchain.
+
+A good estimation of needed amount of gas for publishing a module can be provided by `smove`:
```

Comment:
To estimate the optimal amount of gas for your module, use `smove` tool:

Reply:
Done